### PR TITLE
Create entity classes

### PR DESCRIPTION
**Pull request by maexled**
Created entity classes to work with.
Part of Gamify-IT/issues#164

### CODE REVIEW

File: src/main/java/de/unistuttgart/overworldbackend/data/NPCTask.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package de.unistuttgart.overworldbackend.data;
```
Comment (madebyTimo) [inline src/main/java/de/unistuttgart/overworldbackend/data/NPCTask.java]: Why is it named `NPCTask`? In our Database Model it is named `NPC`

**Comment by madebyTimo** [comment]
I think this PR should be changed into **WIP** because our datamodel still can change due to feedback from our supervisors and the newest version with a `lecture.id` evrerywhere also isn't implemented

**Comment by delvh** [comment]
(I'll review it today during the math lecture. So, expect my review around 1PM.)

File: src/main/java/de/unistuttgart/overworldbackend/controller/LectureController.java [hunk 2]
```diff
@@ -11,16 +20,30 @@
...
+  private LectureRepository lectureRepository;
+
+  @Autowired
+  private LectureService lectureService;
+
+  @Autowired
+  private LectureMapper lectureMapper;
+
   @Operation(summary = "Get all lectures")
   @GetMapping("")
```
Comment (madebyTimo) [inline src/main/java/de/unistuttgart/overworldbackend/controller/LectureController.java:33]:
```suggestion
  @GetMapping("/")
```
Isn't it better to call the mapping `/` cause normally a `/` is prefixed?

Comment (maexled) [reply src/main/java/de/unistuttgart/overworldbackend/controller/LectureController.java:33]: I think it's a normal REST convention not to use a trailing slash

File: src/main/java/de/unistuttgart/overworldbackend/data/NPCTask.java (hunk 1 above)
Comment (maexled) [reply src/main/java/de/unistuttgart/overworldbackend/data/NPCTask.java]: changed to NPC

File: src/main/java/de/unistuttgart/overworldbackend/data/Lecture.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        dungeon.setLecture(this);
+        dungeon.getMinigameTasks().forEach(minigameTask ->
+                minigameTask.setLecture(this)
+        );
+        dungeon.getNpcs().forEach(npc ->
+                npc.setLecture(this)
+        );
+      });
+    });
+  }
```
Comment (Aaron2000119) [inline src/main/java/de/unistuttgart/overworldbackend/data/Lecture.java:58]: The lecutreId gets automatically set in the constructor of the Lecture

File: src/main/java/de/unistuttgart/overworldbackend/controller/LectureController.java (hunk 2 above)
Comment (madebyTimo) [reply src/main/java/de/unistuttgart/overworldbackend/controller/LectureController.java:33]: Okay, I'm not familiar with it. It just seemed strange to me.

**Comment by maexled** [comment]
Closing because it was replaced by #10

**Pull request closed without merging**